src/Identity/Core/src/PasskeyOptions.cs

```diff
@@ -0,0 +1,124 @@
+    /// and <see href="https://www.w3.org/TR/webauthn-3/#dom-publickeycredentialrequestoptions-timeout"/>.
+    /// </remarks>
+    public TimeSpan Timeout { get; set; } = TimeSpan.FromMinutes(5);
+
+    /// <summary>
+    /// The size of the challenge in bytes sent to the client during WebAuthn attestation and assertion.
+    /// </summary>
+    /// <remarks>
+    /// The default value is 32 bytes.
+    /// See <see href="https://www.w3.org/TR/webauthn-3/#dom-publickeycredentialcreationoptions-challenge"/>
+    /// and <see href="https://www.w3.org/TR/webauthn-3/#dom-publickeycredentialrequestoptions-challenge"/>.
+    /// </remarks>
+    public int ChallengeSize { get; set; } = 32;
+
+    /// <summary>
+    /// The effective domain of the server. Should be unique and will be used as the identity for the server.
+    /// </summary>
+    /// <remarks>
+    /// If left <see langword="null"/>, the server's origin may be used instead.
+    /// See <see href="https://www.w3.org/TR/webauthn-3/#rp-id"/>.
+    /// </remarks>
+    public string? ServerDomain { get; set; }
+
+    /// <summary>
+    /// Gets or sets the user verification requirement.
+    /// </summary>
+    /// <remarks>
+    /// See <see href="https://www.w3.org/TR/webauthn-3/#enumdef-userverificationrequirement"/>.
+    /// Possible values are "required", "preferred", and "discouraged".
+    /// The default value is "preferred".
+    /// </remarks>
+    public string? UserVerificationRequirement { get; set; }
+
+    /// <summary>
+    /// Gets or sets the extent to which the server desires to create a client-side discoverable credential.
+    /// Supported values are "discouraged", "preferred", or "required".
+    /// </summary>
+    /// <remarks>
+    /// See <see href="https://www.w3.org/TR/webauthn-3/#enumdef-residentkeyrequirement"/>.
+    /// </remarks>
+    public string? ResidentKeyRequirement { get; set; }
+
+    /// <summary>
+    /// Gets or sets the attestation conveyance preference.
+    /// </summary>
+    /// <remarks>
+    /// See <see href="https://www.w3.org/TR/webauthn-3/#enumdef-attestationconveyancepreference"/>.
+    /// The default value is "none".
+    /// </remarks>
+    public string? AttestationConveyancePreference { get; set; }
+
+    /// <summary>
+    /// Gets or sets the authenticator attachment.
+    /// </summary>
+    /// <remarks>
+    /// See <see href="https://www.w3.org/TR/webauthn-3/#enumdef-authenticatorattachment"/>.
+    /// </remarks>
+    public string? AuthenticatorAttachment { get; set; }
+
+    /// <summary>
+    /// Gets or sets a function that determines whether the given COSE algorithm identifier
+    /// is allowed for passkey operations.
+    /// </summary>
+    /// <remarks>
+    /// If <see langword="null"/> all supported algorithms are allowed.
+    /// See <see href="https://www.iana.org/assignments/cose/cose.xhtml#algorithms"/>.
+    /// </remarks>
+    public Func<int, bool>? IsAllowedAlgorithm { get; set; }
```

Comment:
One alternative to this API would be to allow the developer to specify a list of supported algorithms. However, we don't support all the algorithms anyway, so we'd have to figure out what to do if the developer specifies an algorithm that isn't yet implemented. Also, the `Func` approach allows the developer to specify an "exclude list" if they whish, and if we support more algorithms in the future, the app will automatically support them as well.